polaris/competition/_base.py

```diff
@@ -0,0 +1,360 @@
+from datetime import datetime
+from itertools import chain
+from typing import Callable, TypeAlias, Union
+from typing_extensions import Self
+from loguru import logger
+from pydantic import (
+    Field,
+    FieldSerializationInfo,
+    SerializerFunctionWrapHandler,
+    computed_field,
+    field_serializer,
+    field_validator,
+    model_validator,
+)
+from polaris.dataset._subset import Subset
+from polaris.evaluate import Metric
+from polaris.evaluate._predictions import CompetitionPredictions
+from polaris.experimental._dataset_v2 import DatasetV2
+from polaris.utils.dict2html import dict2html
+from polaris.utils.errors import InvalidCompetitionError
+from polaris.utils.misc import listit
+from polaris.utils.types import (
+    HubOwner,
+    TargetType,
+    TaskType,
+    SplitType,
+    HttpUrlString,
+    SlugCompatibleStringType,
+    PredictionsType,
+    HubUser,
+)
+
+ColumnsType: TypeAlias = str | list[str]
+
+
+class CompetitionSpecification(DatasetV2):
```

Comment:
**Question**
Would it make sense to have this inherit from `BenchmarkSpecification` (or even the new `BenchmarkV1Specification`)? It's basically a combination of dataset and benchmark.

Reply:
Yes! It may requires some shuffling of what's part of the base class and what's not, but I do think that is the better approach here. How do you feel about inheriting the V2 benchmark already?